Does not build with musl

<details>

Make sure that these boxes are checked before submitting your issue -- thank you!

- [x] This issue is for the C core of igraph.
- [x] This issue is a bug report or a feature request, not a support question.

</details>

here you define `OFF_T` as `__off64_t` when building under linux, but when building with musl, it doesn’t exist:

https://github.com/igraph/igraph/blob/02005680aebb4281f2964c6d2d8d98851fce63b9/src/f2c/sysdep1.h#L16-L18

musl [defines](https://github.com/runtimejs/musl-libc/blob/master/include/sys/types.h#L83) `off64_t` instead of `__off64_t`, so the check needs to be different than `__linux__` or `__sun__`.

OK, I circumvented it with the hacky ` CC='gcc -D__off64_t=off64_t' pip3 install python-igraph`. Now I run in the second assumption problem:

    f2c/uninit.c:276:25: fatal error: fpu_control.h: No such file or directory

--- a/src/f2c/sysdep1.h
+++ b/src/f2c/sysdep1.h
@@ -15,8 +15,13 @@
 
 #ifdef __linux__
 #define USE_LARGEFILE
+
+#ifdef __GLIBC__
 #define OFF_T __off64_t
-#endif
+#else
+#define OFF_T off64_t
+#endif /* __GLIBC__ */
+#endif /* __linux__ */
 
 #ifdef _AIX43
 #define _LARGE_FILES

--- a/src/f2c/uninit.c
+++ b/src/f2c/uninit.c
@@ -250,9 +250,11 @@ ieee0(Void)
  */
 
 
-#if (defined(__GLIBC__)&& ( __GLIBC__>=2) && (__GLIBC_MINOR__>=2) )
-#define _GNU_SOURCE 1
+#ifdef __GLIBC__
 #define IEEE0_done
+
+#if ((__GLIBC__>=2) && (__GLIBC_MINOR__>=2))
+#define _GNU_SOURCE 1
 #include <fenv.h>
  static void
   ieee0(Void)
@@ -265,14 +267,11 @@ ieee0(Void)
          unsupported_error();
 }
 
-#endif /* Glibc control */
-
 /* Many linux cases will be treated through GLIBC.  Note that modern
  * linux runs on many non-i86 plaforms and as a result the following code
  * must be processor dependent rather than simply OS specific */
 
-#if (defined(__linux__)&&(!defined(IEEE0_done)))
-#define IEEE0_done
+#else /* __GLIBC__<2.2 */
 #include <fpu_control.h>
 
 
@@ -401,7 +400,8 @@ static void ieee0(Void)
 
 #endif /* _FPU_IEEE */
 	}
-#endif /* __linux__ */
+#endif /* __GLIBC__>2.2 */
+#endif /* __GLIBC__ */
 
 /* Specific to OSF/1 */
 #if (defined(__alpha)&&defined(__osf__))

